core/src/main/java/org/apache/iceberg/DataTableScan.java

```diff
@@ -81,8 +81,8 @@ public CloseableIterable<FileScanTask> doPlanFiles() {
     FileIO io = table().io();
     List<ManifestFile> dataManifests = snapshot.dataManifests(io);
     List<ManifestFile> deleteManifests = snapshot.deleteManifests(io);
-    scanMetrics().totalDataManifests().increment(dataManifests.size());
-    scanMetrics().totalDeleteManifests().increment(deleteManifests.size());
+    scanMetrics().totalDataManifests().increment((long) dataManifests.size());
+    scanMetrics().totalDeleteManifests().increment((long) deleteManifests.size());
```

Comment:
Should we add `increment(int)` to avoid needing to add casts?

Reply:
good point, added as part of https://github.com/apache/iceberg/pull/5505